packages/edit-site/src/store/selectors.js

```diff
@@ -31,16 +33,35 @@ import {
  * @typedef {'template'|'template_type'} TemplateType Template type.
  */
 
+/**
+ * Helper for getting a preference from the preferences store.
+ *
+ * This is only present so that `getSettings` doesn't need to be made a
+ * registry selector.
+ *
+ * It's unstable because the selector needs to be exported and so part of the
+ * public API to work.
+ */
+export const __unstableGetPreference = createRegistrySelector(
+	( select ) => ( state, name ) =>
+		select( preferencesStore ).get( 'core/edit-site', name )
+);
```

Comment:
This is the part I'm most unhappy with. Ideally any helper is internal only, but the registry selector needs to be exported to work.

The separate selector is also needed for the memoized `getSettings` selector.

If there's a different way to do this that anyone knows of, that'd be great.

Reply:
Hm, yeah, that's an annoying limitation of `createRegistrySelector`.

I think we may just have to inline this function so that `isFeatureActive`, `getSettings` are registry selectors that call `select( preferencesStore ).get` directly.

Unless @adamziel or @jsnajdr can think of something better.

Reply:
> I think we may just have to inline this function so that `isFeatureActive`, `getSettings` are registry selectors that call `select( preferencesStore ).get` directly.

If you insist on having a `getPreference` for code re-use (I'd argue it's maybe not worth it since it's only one line) then `getPreference` could accept `select` as an argument.

```js
const getPreference = ( select, name ) =>
	select( preferencesStore ).get( 'core/edit-site', name );
export const isFeatureActive = createRegistrySelector(
	( select ) => ( state, featureName ) => {
		return !! getPreference( select, featureName );
	}
);
```

Reply:
The thing that makes it difficult is the second argument to `createSelector` (which is used for memoizing `getSettings`) is one of the places `__unstableGetPreference` needs to be used. That has no way to access `select` from what I can see.

I suppose it would be possible to import `select` from the data store for this purpose. I don't know whether that's an ok practice.

Reply:
I'm afraid there are only two options available:
- export the `__unstableGetPreference` selector as part of the public API. The only reason we need to do that is that the code that initializes the public selectors [assigns to the `selector.registry` field](https://github.com/WordPress/gutenberg/blob/trunk/packages/data/src/redux-store/index.js#L144).
- wrap all the selectors that use `__unstableGetPreference` with `createRegistrySelector` and inline the helper.

> import `select` from the data store for this purpose

That would mean that the store can be ever registered only with the default registry and never any other one. Because such an import would hardcode a reference to the default registry to the selector.

`createRegistrySelector` will already fail when you try to register its store with two registries -- because they will fight over the `selector.registry` field, both assigning to them. We'll need to rethink the entire concept at some point.

Reply:
> The thing that makes it difficult is the second argument to `createSelector`

Ah nuts 🥜

> We'll need to rethink the entire concept at some point.

Yeah. I wonder if we can do something similar to thunks where selectors (and the second argument of `createSelector`) can return a function which is passed an object containing `registry`, `select`, etc.

Reply:
> That would mean that the store can be ever registered only with the default registry and never any other one. Because such an import would hardcode a reference to the default registry to the selector.

I see. I remember there being a reason for avoiding this practice, but couldn't remember what it was.

Seems like making it `__unstable` is probably the best trade-off for now. Thanks for all the feedback.